### PR TITLE
refactor(behavior_path_planner): separate drivable area functions

### PR DESCRIPTION
**Pull request by zulfaqar-azmi-t4**
## Description

Caution: Although this is not breaking changes, it still will create a lot of conflict for the present PR or during creation of experimental branch. 

Previously, drivable area expansion algorithm was written in the `utils/utils.hpp`.
With the new dynamic drivable area expansion feature added in the behavior path planner module, there is a necessity to separate the algorithm from `utils.hpp` into another file to prevent confusion when debugging drivable area related functions.

## Related links

None

## Tests performed

Compile success

## Notes for reviewers

None

## Interface changes

None

## Effects on system behavior

None

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [ ] I've confirmed the [contribution guidelines].
- [ ] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].
- [x] The PR has been properly tested.
- [x] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_path_planner/include/behavior_path_planner/utils/drivable_area_expansion/static_drivable_area.hpp [hunk 1]
```diff
@@ -0,0 +1,99 @@
+// Copyright 2021-2023 Tier IV, Inc.
```
Comment (satoshi-ota) [inline planning/behavior_path_planner/include/behavior_path_planner/utils/drivable_area_expansion/static_drivable_area.hpp:1]: TIER IV :policeman: 

```suggestion
// Copyright 2023 TIER IV, Inc.
```

Comment (zulfaqar-azmi-t4) [reply planning/behavior_path_planner/include/behavior_path_planner/utils/drivable_area_expansion/static_drivable_area.hpp]: Haha, I always missed that.

**Review by satoshi-ota — APPROVED** [review]
LGTM!

**Comment by satoshi-ota** [comment]
@kosuke55 How about is your review from goal planner side...?

**Review by kosuke55 — APPROVED** [review]
looks good!! thanks

**Pull request merged**